src/MockHttpClient/HttpRequestMessageExtensions.cs

```diff
@@ -130,7 +132,104 @@ public static bool HasHeader(this HttpRequestMessage request, string headerName,
 
             return false;
         }
-    }
 
+        /// <summary>
+        /// Determines whether the request contains the specified content as string.
+        /// </summary>
+        /// <param name="request">the request</param>
+        /// <param name="expectedContent">the expected content</param>
+        /// <returns></returns>
+        public static bool HasStringContent(this HttpRequestMessage request, string expectedContent)
+        {
+            if (request.Content == null)
+            {
+                throw new ArgumentException($"string content '{expectedContent}' was expected but content is null");
+            }
 
+            var content = request.Content.ReadAsStringAsync().Result;
```

Comment:
Unfortunate we have to synchronize the async

Would be good to setup `When(Func<HttpRequsetMessage, Task<bool>> condition)` so that we can handle async conditions. But that change might be too involved for what you trying to achieve with this PR

If we are going to synchronize it use `request.Content.ReadAsStringAsync().GetAwaiter().GetResult()` instead of `.Result` . It handles exceptions a lot nicer

Reply:
Thanks for the hint. Good catch.